<!DOCTYPE html>
<html lang="en">

	<!-- Head -->
	<head>

		<!-- Basic Page Needs
		–––––––––––––––––––––––––––––––––––––––––––––––––– -->
		<meta charset="utf-8">
		<title>David B. Blumenthal</title>
		<meta name="description" content="David B. Blumenthal's personal homepage.">
		<meta name="author" content="David B. Blumenthal">

		<!-- Mobile Specific Metas
		–––––––––––––––––––––––––––––––––––––––––––––––––– -->
		<meta name="viewport" content="width=device-width, initial-scale=1">

		<!-- FONT
		–––––––––––––––––––––––––––––––––––––––––––––––––– -->
		<link href='https://fonts.googleapis.com/css?family=Raleway:400,300,600' rel='stylesheet' type='text/css'>

		<!-- CSS
		–––––––––––––––––––––––––––––––––––––––––––––––––– -->
		<link rel="stylesheet" href="css/normalize.css">
		<link rel="stylesheet" href="css/skeleton.css">
		<link rel="stylesheet" href="css/custom.css">

		<!-- Scripts
		–––––––––––––––––––––––––––––––––––––––––––––––––– -->
		<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
		<script src="js/site.js"></script>

	</head>

	<!-- Body -->
	<body class="code-snippets-visible">

		<!-- Primary Page Layout
		–––––––––––––––––––––––––––––––––––––––––––––––––– -->
		<div class="container" id="top">

			<!-- Header -->
			<section class="header">
				<h2 class="title">Dr. David B. Blumenthal</h2>
				<p>
					Chair of Experimental Bioinformatics<br>
					TUM School of Life Sciences<br>
					Technical University of Munich
				</p>
				<p>
					<script type="text/javascript">
						var email="david.blumenthal"
						var domain="wzw.tum.de"
						document.write("<a href="+"mail"+"to:"+email+"@"+domain+">"+"[firstname].[lastname]@wzw.tum.de"+"<\/a>")
					</script><br>
					<a itemprop="sameAs" content="https://orcid.org/0000-0001-8651-750X" href="https://orcid.org/0000-0001-8651-750X" target="orcid.widget" rel="noopener noreferrer" style="vertical-align:top;"><img src="https://orcid.org/sites/default/files/images/orcid_16x16.png" style="width:1em;margin-right:.5em;" alt="ORCID iD icon">orcid.org/0000-0001-8651-750X</a><br>
					<a itemprop="sameAs" content="https://github.com/dbblumenthal" href="https://github.com/dbblumenthal" target="github.widget" rel="noopener noreferrer" style="vertical-align:top;"><img src="img/github.png" style="width:1em;margin-right:.5em;" alt="GitHub mark">github.com/dbblumenthal</a><br>
				</p>
			</section>

			<!-- Navigation bar -->
			<div class="navbar-spacer"></div>
			<nav class="navbar">
				<div class="container">
					<ul class="navbar-list">
						<li class="navbar-item navbar-desktop">
							<a class="navbar-link" href="#top">About Me</a>
						</li>
						<li class="navbar-item navbar-desktop">
							<a class="navbar-link" href="#publications">Publications</a>
						</li>
						<li class="navbar-item navbar-desktop">
							<a class="navbar-link" href="#teaching">Teaching</a>
						</li>
						<li class="navbar-item navbar-desktop">
							<a class="navbar-link" href="#software">Software</a>
						</li>
						<li class="navbar-item navbar-desktop">
							<a class="navbar-link" href="#awards">Awards & Funding</a>
						</li>
						<li class="navbar-item navbar-desktop">
							<a class="navbar-link" href="#services">Services</a>
						</li>
						<li class="navbar-item navbar-mobile">
							<a class="navbar-link navbar-mobile-link" href="#" data-popover="#mobileNavPopover">Menu</a>
							<div id="mobileNavPopover" class="popover">
								<ul class="popover-list">
									<li class="popover-item">
										<a class="popover-link" href="#top">About Me</a>
									</li>
									<li class="popover-item">
										<a class="popover-link" href="#publications">Publications</a>
									</li>
									<li class="popover-item">
										<a class="popover-link" href="#teaching">Teaching</a>
									</li>
									<li class="popover-item">
										<a class="popover-link" href="#software">Software</a>
									</li>
									<li class="popover-item">
										<a class="popover-link" href="#awards">Awards & Funding</a>
									</li>
									<li class="popover-item">
										<a class="popover-link" href="#services">Services</a>
									</li>
								</ul>
							</div>
						</li>
					</ul>
				</div>
			</nav>

			<!-- About Me -->
			<div class="docs-section" id="about-me">
				<h6 class="docs-header">About Me</h6>
				<div class="row">
					<div class="five columns">
						<img class="bio-pic" src="img/photo.jpg" height="220">
					</div>
					<div class="seven columns left-align">
						<p class="bio">
							I am a postdoctoral fellow at the <a href="https://www.exbio.de" target="_blank">Chair of Experimental Bioinformatics</a> of the <a href="https://www.tum.de/" target="_blank">Technical University of Munich</a>, where I am developing new algorithmic techniques for network biology and systems medicine. In July 2019, I defended my Ph.D. thesis entitled &ldquo;New Techniques for Graph Edit Distance Computation&rdquo; at the <a href="https://www.unibz.it/" target="_blank">Free University of Bozen-Bolzano</a>. Before moving to Bolzano, I studied mathematics and philosophy at the <a href="http://www.fu-berlin.de/" target="_blank">Freie Universit&auml;t Berlin</a> and at the <a href="http://www.tu-berlin.de/" target="_blank">Technische Universit&auml;t Berlin</a>. For more information, please download my <a href="res/blumenthal_cv.pdf" target="_blank">CV</a>.
						</p>
					</div>
				</div>
			</div>

			<!-- Publications -->
			<div class="docs-section" id="publications">
				<h6 class="docs-header">Publications</h6>
				<p class="left-align">
					This is a complete list of my research publications. Also see my profiles on <a href="https://www.researchgate.net/profile/David_Blumenthal3" target="_blank">ResearchGate</a>, <a href="https://orcid.org/0000-0001-8651-750X" target="_blank">ORCiD</a>, <a href="https://dblp.uni-trier.de/pers/hd/b/Blumenthal:David_B=" target="_blank">dblp</a>, and <a href="https://scholar.google.com/citations?user=B3DmHU4AAAAJ&hl=en" target="_blank">Google Scholar</a>. If you have problems accessing a paper you are interested in, feel free to contact me.
				</p>
				<!-- Bioinf. 2020b -->
				<div class="wrap-collabsible left-align">
					<input id="bioinf2020b" class="toggle" type="checkbox">
					<label for="bioinf2020b" class="lbl-toggle">
						<strong>[2020]</strong> D. B. Blumenthal, J. Baumbach, M. Hoffmann, T. Kacprowski, M. List:
						&ldquo;A framework for modeling epistatic interaction&rdquo;,
						<strong>Bioinformatics</strong>, in press.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> 
								<p>
									<strong>Motivation:</strong>
									Recently, various tools for detecting single nucleotide polymorphisms (SNPs) involved in epistasis have been developed. However, no studies evaluate the employed statistical epistasis models such as the &chi;<sup>2</sup>-test or quadratic regression independently of the tools that use them. Such an independent evaluation is crucial for developing improved epistasis detection tools, for it allows to decide if a tool's performance should be attributed to the epistasis model or to the optimization strategy run on top of it.
								</p>
								<p>
									<strong>Results:</strong>
									We present a protocol for evaluating epistasis models independently of the tools they are used in and generalize existing models designed for dichotomous phenotypes to the categorical and quantitative case. Additionally, we propose a new model which scores candidate SNP sets by computing maximum likelihood distributions for the observed phenotypes in the cells of their penetrance tables. Extensive experiments show that the proposed maximum likelihood model outperforms three widely used epistasis models in most cases. The experiments also provide valuable insights into the properties of existing models, for instance, that quadratic regression perform particularly well on instances with quantitative phenotypes.
								</p>
								<p>
									<strong>Availability and implementation:</strong>
									The evaluation protocol and all compared models are implemented in C++ and are supported under Linux and macOS. They are available at <a href="https://github.com/baumbachlab/genepiseeker" target="_blank">https://github.com/baumbachlab/genepiseeker</a>, along with test datasets and scripts to reproduce the experiments.
								</p>
								<p>
									<strong>Supplementary information:</strong>
									Supplementary information is available at <i>Bioinformatics</i> online.
								</p>
							</p>
							<p>
								<a href="https://doi.org/10.1093/bioinformatics/btaa990" target="_blank">doi</a> | <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaa245/5820008?guestAccessKey=2385c5f1-6587-4065-b2f2-1039765fa5b6" target="_blank">paper</a> | <a href="res/bioinf2020b.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- ASSAY 2020 -->
				<div class="wrap-collabsible left-align">
					<input id="assay2020" class="toggle" type="checkbox">
					<label for="assay2020" class="lbl-toggle">
						<strong>[2020]</strong> J. Matschinske, M. Salgado-Albarrán, S. Sadegh, D. Bongiovanni, J. Baumbach, D. B. Blumenthal:
						&ldquo;Individuating possibly repurposable drugs and drug targets for COVID-19 treatment via hypothesis-driven systems medicine using CoVex&rdquo;,
						<strong>Assay Drug Dev. Technol.</strong>, in press.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> Coronavirus disease 2019 (COVID-19), caused by the SARS-CoV-2 virus, has developed into a pandemic causing major disruptions and hundreds of thousands of deaths in wide parts of the world. As of July 3, 2020, neither vaccines nor approved drugs for effective treatment are available. In this paper, we showcase how to individuate drug targets and potentially repurposable drugs in silico using CoVex (<a href="https://exbio.wzw.tum.de/covex/" target="_blank">https://exbio.wzw.tum.de/covex/</a>), a recently presented systems medicine platform for COVID-19 drug repurposing. Starting from initial hypotheses, CoVex leverages network algorithms to individuate host proteins involved in COVID-19 disease mechanisms, as well as existing drugs targeting these potential drug targets. Our analysis reveals GLA, PLAT, and GGCX as potential drug targets, and urokinase, argatroban, dabigatran etexilate, betrixaban, ximelagatran and anisindione as potentially repurposable drugs.
							</p>
							<p>
								<a href="https://doi.org/10.1089/adt.2020.1010" target="_blank">doi</a> | <a href="res/assay2020.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- SCIM 2020 -->
				<div class="wrap-collabsible left-align">
					<input id="scim2020" class="toggle" type="checkbox">
					<label for="scim2020" class="lbl-toggle">
						<strong>[2020]</strong> S. Helmer, D. B. Blumenthal, K. Paschen:
						&ldquo;What is meaningful research and how should we measure it?&rdquo;,
						<strong>Scientometrics 125</strong>, pp. 153-169.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> We discuss the trend towards using quantitative metrics for evaluating research. We claim that, rather than promoting meaningful research, purely metric-based research evaluation schemes potentially lead to a dystopian academic reality, leaving no space for creativity and intellectual initiative. After sketching what the future could look like if quantitative metrics are allowed to proliferate, we provide a more detailed discussion on why research is so difficult to evaluate and outline approaches for avoiding such a situation. In particular, we characterize meaningful research as an essentially contested concept and argue that quantitative metrics should always be accompanied by operationalized instructions for their proper use  and continuously evaluated via feedback loops. Additionally, we analyze a dataset containing information about computer science publications and their citation history and indicate how quantitative metrics could potentially be calibrated via alternative evaluation methods such as test of time awards. Finally, we argue that, instead of over-relying on indicators, research environments should primarily be based on trust and personal responsibility.
							</p>
							<p>
								<a href="https://doi.org/10.1007/s11192-020-03649-5" target="_blank">doi</a> | <a href="https://link.springer.com/content/pdf/10.1007/s11192-020-03649-5.pdf" target="_blank">paper</a> | <a href="res/scim2020.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- Nat. Comm. 2020 -->
				<div class="wrap-collabsible left-align">
					<input id="natcommun2020" class="toggle" type="checkbox">
					<label for="natcommun2020" class="lbl-toggle">
						<strong>[2020]</strong> S. Sadegh, J. Matschinske, D. B. Blumenthal, G. Galindez, T. Kacprowski, M. List, R. Nasirigerdeh, M. Oubounyt, A. Pichlmair, T. D. Rose, M. Salgado-Albarrán, J. Späth, A. Stukalov, N. K. Wenke, K. Yuan, J. K. Pauling, J. Baumbach:
						&ldquo;Exploring the SARS-CoV-2 virus-host-drug interactome for drug repurposing&rdquo;,
						<strong>Nat. Commun. 11(1)</strong>, pp. 3518:1-3518:9.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> Coronavirus Disease-2019 (COVID-19) is an infectious disease caused by the SARS-CoV-2 virus. Various studies exist about the molecular mechanisms of viral infection. However, such information is spread across many publications and it is very time-consuming to integrate, and exploit. We develop CoVex, an interactive online platform for SARS-CoV-2 host interactome exploration and drug (target) identification. CoVex integrates virus-human protein interactions, human protein-protein interactions, and drug-target interactions. It allows visual exploration of the virus-host interactome and implements systems medicine algorithms for network-based prediction of drug candidates. Thus, CoVex is a resource to understand molecular mechanisms of pathogenicity and to prioritize candidate therapeutics. We investigate recent hypotheses on a systems biology level to explore mechanistic virus life cycle drivers, and to extract drug repurposing candidates. CoVex renders COVID-19 drug research systems-medicine-ready by giving the scientific community direct access to network medicine algorithms. It is available at <a href="https://exbio.wzw.tum.de/covex/" target="_blank">https://exbio.wzw.tum.de/covex/<a>.
							</p>
							<p>
								<a href="https://doi.org/10.1038/s41467-020-17189-2" target="_blank">doi</a> | <a href="https://www.nature.com/articles/s41467-020-17189-2.pdf" target="_blank">paper</a> | <a href="res/natcommun2020.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- Bioinf. 2020a -->
				<div class="wrap-collabsible left-align">
					<input id="bioinf2020a" class="toggle" type="checkbox">
					<label for="bioinf2020a" class="lbl-toggle">
						<strong>[2020]</strong> D. B. Blumenthal, L. Viola, M. List, J. Baumbach, P. Tieri, T. Kacprowski:
						&ldquo;EpiGEN: an epistasis simulation pipeline&rdquo;,
						<strong>Bioinformatics</strong>, in press.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> 
								<p>
									<strong>Summary:</strong>
									Simulated data is crucial for evaluating epistasis detection tools in genome-wide association studies. Existing simulators are limited, as they do not account for linkage disequilibrium (LD), support limited interaction models of single nucleotide polymorphisms (SNPs) and only dichotomous phenotypes, or depend on proprietary software. In contrast, EpiGEN supports SNP interactions of arbitrary order, produces realistic LD patterns, and can generate both categorical and quantitative phenotypes.
								</p>
								<p>
									<strong>Availability and implementation:</strong>
									EpiGEN is implemented in Python 3 and is freely available at <a href="https://github.com/baumbachlab/epigen" target="_blank">https://github.com/baumbachlab/epigen</a>.
								</p>
								<p>
									<strong>Supplementary information:</strong>
									A detailed user guide for EpiGEN is available at <i>Bioinformatics</i> online.
								</p>
							</p>
							<p>
								<a href="https://doi.org/10.1093/bioinformatics/btaa245" target="_blank">doi</a> | <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btaa245/5820008?guestAccessKey=2385c5f1-6587-4065-b2f2-1039765fa5b6" target="_blank">paper</a> | <a href="res/bioinf2020a.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- VLDB J. 2020b -->
				<div class="wrap-collabsible left-align">
					<input id="vldbj2020b" class="toggle" type="checkbox">
					<label for="vldbj2020b" class="lbl-toggle">
						<strong>[2020]</strong> T. Chondrogiannis, P. Bouros, J. Gamper, U. Leser, D. B. Blumenthal:
						&ldquo;Finding k-shortest paths with limited overlap&rdquo;,
						<strong>VLDB J. 29</strong>, pp. 1023-1047.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> In this paper, we investigate the computation of alternative paths between two locations in a road network. More specifically, we study the k-shortest paths with limited overlap (kSPwLO) problem that aims at finding a set of k paths such that all paths are sufficiently dissimilar to each other and as short as possible. To compute kSPwLO queries, we propose two exact algorithms, termed OnePass and MultiPass, and we formally prove that MultiPass is optimal in terms of complexity. We also study two classes of heuristic algorithms: (a) performance-oriented heuristic algorithms that trade shortness for performance, i.e., they reduce query processing time, but do not guarantee that the length of each subsequent result is minimum; and (b) completeness-oriented heuristic algorithms that trade dissimilarity for completeness, i.e., they relax the similarity constraint to return a result that contains exactly k paths. An extensive experimental analysis on real road networks demonstrates the efficiency of our proposed solutions in terms of runtime and quality of the result. 
							</p>
							<p>
								<a href="https://doi.org/10.1007/s00778-020-00604-x" target="_blank">doi</a> | <a href="res/vldbj2020b.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- VLDB J. 2020a -->
				<div class="wrap-collabsible left-align">
					<input id="vldbj2020a" class="toggle" type="checkbox">
					<label for="vldbj2020a" class="lbl-toggle">
						<strong>[2020]</strong> D. B. Blumenthal, N. Boria, J. Gamper, S. Bougleux, L. Brun:
						&ldquo;Comparing heuristics for graph edit distance computation&rdquo;,
						<strong>VLDB J. 29</strong>, pp. 419-458.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> Because of its flexibility, intuitiveness, and expressivity, the graph edit distance (GED) is one of the most widely used distance measures for labeled graphs. Since exactly computing GED is NP-hard, over the past years, various heuristics have been proposed. They use techniques such as transformations to the linear sum assignment problem with error-correction, local search, and linear programming to approximate GED via upper or lower bounds. In this paper, we provide a systematic overview of the most important heuristics. Moreover, we empirically evaluate all compared heuristics within an integrated implementation.
							</p>
							<p>
								<a href="https://doi.org/10.1007/s00778-019-00544-1" target="_blank">doi</a> | <a href="res/vldbj2020a-poster.pdf" target="_blank">poster</a> | <a href="res/vldbj2020a.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- PRL 2020 (Improved local search ...)-->
				<div class="wrap-collabsible left-align">
					<input id="prl2020c" class="toggle" type="checkbox">
					<label for="prl2020c" class="lbl-toggle">
						<strong>[2020]</strong> N. Boria, D. B. Blumenthal, S. Bougleux, L. Brun:
						&ldquo;Improved local search for graph edit distance&rdquo;,
						<strong>Pattern Recognit. Lett. 129</strong>, pp. 19-25.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> The graph edit distance (GED) measures the dissimilarity between two graphs as the minimal cost of a sequence of elementary operations transforming one graph into another. This measure is fundamental in many areas such as structural pattern recognition or classification. However, exactly computing GED is NP-hard. Among different classes of heuristic algorithms that were proposed to compute approximate solutions, local search based algorithms provide the tightest upper bounds for GED. In this paper, we present K-REFINE and RANDPOST. K-REFINE generalizes and improves an existing local search algorithm and performs particularly well on small graphs. RANDPOST is a general warm start framework that stochastically generates promising initial solutions to be used by any local search based GED algorithm. It is particularly efficient on large graphs. An extensive empirical evaluation demonstrates that both K-REFINE and RANDPOST perform excellently in practice.
							</p>
							<p>
								<a href="https://doi.org/10.1016/j.patrec.2019.10.028" target="_blank">doi</a> | <a href="res/prl2020c.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- PRL 2020 (On the exact ...)-->
				<div class="wrap-collabsible left-align">
					<input id="prl2020a" class="toggle" type="checkbox">
					<label for="prl2020a" class="lbl-toggle">
						<strong>[2020]</strong> D. B. Blumenthal, J. Gamper:
						&ldquo;On the exact computation of the graph edit distance&rdquo;,
						<strong>Pattern Recognit. Lett. 134</strong>, pp. 46-57.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> The graph edit distance is a widely used distance measure for labelled graph. However, <span class="typewriter">A&#42;-GED</span>, the standard approach for its exact computation, suffers from huge runtime and memory requirements. Recently, three better performing algorithms have been proposed: The general algorithms <span class="typewriter">DF-GED</span> and <span class="typewriter">BIP-GED</span>, and the algorithm <span class="typewriter">CSI-GED</span>, which only works for uniform edit costs. All newly proposed algorithms outperform the standard approach <span class="typewriter">A&#42;-GED</span>. However, cross-comparisons are lacking. This paper consolidates and extends these recent advances. To this purpose, we present all existing algorithms in a unified way and show that the slightly different definitions of the graph edit distance underlying <span class="typewriter">A&#42;-GED</span> and <span class="typewriter">DF-GED</span>, on the one side, and <span class="typewriter">CSI-GED</span>, on the other side, can be harmonised. This harmonisation allows us to develop a generalisation of <span class="typewriter">CSI-GED</span> to non-uniform edit cost. Moreover, we present a speed-up of <span class="typewriter">A&#42;-GED</span> and <span class="typewriter">DF-GED</span> for uniform edit costs, which build upon the fact that, in the uniform case, a continuously used subroutine can be implemented to run in linear rather than cubic time. We also suggest an algorithm <span class="typewriter">MIP-GED</span> which builds upon a very compact new mixed integer linear programming formulation. Finally, we carry out a thorough empirical evaluation, which, for the first time, compares all existing exact algorithms.
							</p>
							<p>
								<a href="https://doi.org/10.1016/j.patrec.2018.05.002" target="_blank">doi</a> | <a href="res/prl2020a.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div> 
				<!-- PRL 2020 (Fast linear sum ...)-->
				<div class="wrap-collabsible left-align">
					<input id="prl2020b" class="toggle" type="checkbox">
					<label for="prl2020b" class="lbl-toggle">
						<strong>[2020]</strong> S. Bougleux, B. Ga&uuml;z&egrave;re, D. B. Blumenthal, L. Brun:
						&ldquo;Fast linear sum assignment with error-correction and no cost constraints&rdquo;,
						<strong>Pattern Recognit. Lett. 134</strong>, pp. 37-45.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> We propose an algorithm that efficiently solves the linear sum assignment problem with error-correction and no cost constraints. This problem is encountered for instance in the approximation of the graph edit distance. The fastest currently available solvers for the linear sum assignment problem require the pairwise costs to respect the triangle inequality. Our algorithm is as fast as these algorithms, but manages to drop the cost constraint. The main technical ingredient of our algorithm is a cost-dependent factorization of the node substitutions.
							</p>
							<p>
								<a href="https://doi.org/10.1016/j.patrec.2018.03.032" target="_blank">doi</a> | <a href="res/prl2020b.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- PhD Thesis -->
				<div class="wrap-collabsible left-align">
					<input id="thesis" class="toggle" type="checkbox">
					<label for="thesis" class="lbl-toggle">
						<strong>[2019]</strong> D. B. Blumenthal:
						&ldquo;New Techniques for Graph Edit Distance Computation&rdquo;,
						<strong>Ph.D. Thesis</strong>, Free University of Bozen-Bolzano.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> Due to their capacity to encode rich structural information, labeled graphs are often used for modeling various kinds of objects such as images, molecules, and chemical compounds. If pattern recognition problems such as clustering and classification are to be solved on these domains, a (dis-)similarity measure for labeled graphs has to be defined. A widely used measure is the graph edit distance (GED), which, intuitively, is defined as the minimum amount of distortion that has to be applied to a source graph in order to transform it into a target graph. The main advantage of GED is its flexibility and sensitivity to small differences between the input graphs. Its main drawback is that it is hard to compute.</p><p>In this thesis, new results and techniques for several aspects of computing GED are presented. Firstly, theoretical aspects are discussed: competing definitions of GED are harmonized, the problem of computing GED is characterized in terms of complexity, and several reductions from GED to the quadratic assignment problem (QAP) are presented. Secondly, solvers for the linear sum assignment problem with error-correction (LSAPE) are discussed. LSAPE is a generalization of the well-known linear sum assignment problem (LSAP), and has to be solved as a subproblem by many GED algorithms. In particular, a new solver is presented that efficiently reduces LSAPE to LSAP. Thirdly, exact algorithms for computing GED are presented in a systematic way, and improvements of existing algorithms as well as a new mixed integer programming (MIP) based approach are introduced. Fourthly, a detailed overview of heuristic algorithms that approximate GED via upper and lower bounds is provided, and eight new heuristics are described. Finally, a new easily extensible C++ library for exactly or approximately computing GED is presented.
							</p>
							<p>
								<a href="https://arxiv.org/abs/1908.00265" target="_blank">arXiv</a> | <a href="res/thesis.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- GbRPR 2019 -->
				<div class="wrap-collabsible left-align">
					<input id="gbrpr2019" class="toggle" type="checkbox">
					<label for="gbrpr2019" class="lbl-toggle">
						<strong>[2019]</strong> D. B. Blumenthal, S. Bougleux, J. Gamper, L. Brun:
						&ldquo;GEDLIB: A C++ library for graph edit distance computation&rdquo;,
						<strong>GbRPR 2019</strong>, pp. 14-24.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> The graph edit distance (GED) is a flexible graph dissimilar- ity measure widely used within the structural pattern recognition field. In this paper, we present GEDLIB, a C++ library for exactly or approx- imately computing GED. Many existing algorithms for GED are already implemented in GEDLIB. Moreover, GEDLIB is designed to be easily ex- tensible: for implementing new edit cost functions and GED algorithms, it suffices to implement abstract classes contained in the library. For implementing these extensions, the user has access to a wide range of utilities, such as deep neural networks, support vector machines, mixed integer linear programming solvers, a blackbox optimizer, and solvers for the linear sum assignment problem with and without error-correction.
							</p>
							<p>
								<a href="https://doi.org/10.1007/978-3-030-20081-7_2" target="_blank">doi</a> | <a href="res/gbrpr2019-slides.pdf" target="_blank">slides</a> | <a href="res/gbrpr2019-poster.pdf" target="_blank">poster</a> | <a href="res/gbrpr2019.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- SIGSPATIAL 2018 -->
				<div class="wrap-collabsible left-align">
					<input id="sigspatial2018" class="toggle" type="checkbox">
					<label for="sigspatial2018" class="lbl-toggle">
						<strong>[2018]</strong> T. Chondrogiannis, P. Bouros, J. Gamper, U. Leser, D. B. Blumenthal:
						&ldquo;Finding k-dissimilar paths with minimum collective length&rdquo;,
						<strong>SIGSPATIAL 2018</strong>, pp. 404-407.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> Shortest path computation is a fundamental problem in road networks. However, in many real-world scenarios, determining solely the shortest path is not enough. In this paper, we study the problem of finding k-Dissimilar Paths with Minimum Collective Length (kDPwML), which aims at computing a set of paths from a source s to a target t such that all paths are pairwise dissimilar by at least \theta and the sum of the path lengths is minimal. We introduce an exact algorithm for the kDPwML problem, which iterates over all possible s-t paths while employing two pruning techniques to reduce the prohibitively expensive computational cost. To achieve scalability, we also define the much smaller set of the simple single-via paths, and we adapt two algorithms for kDPwML queries to iterate over this set. Our experimental analysis on real road networks shows that iterating over all paths is impractical, while iterating over the set of simple single-via paths can lead to scalable solutions with only a small trade-off in the quality of the results.
							</p>
							<p>
								<a href="https://doi.org/10.1145/3274895.3274903" target="_blank">doi</a> | <a href="res/sigspatial2018.bib" target="_blank">bibtex</a> | <a href="https://arxiv.org/abs/1809.06831" target="_blank">extended version on arXiv</a>
							</p>
						</div>
					</div>
				</div>
				<!-- ICPR 2018 -->
				<div class="wrap-collabsible left-align">
					<input id="icpr2018" class="toggle" type="checkbox">
					<label for="icpr2018" class="lbl-toggle">
						<strong>[2018]</strong> D. B. Blumenthal, &Eacute;. Daller, S. Bougleux, L. Brun, J. Gamper:
						&ldquo;Quasimetric graph edit distance as a compact quadratic assignment problem&rdquo;,
						<strong>ICPR 2018</strong>, pp. 934-939.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> The graph edit distance (GED) is a widely used distance measure for attributed graphs. It has recently been shown that the problem of computing GED, which is a NP-hard optimization problem, can be formulated as a quadratic assignment problem (QAP). This formulation is useful, since it allows to derive well performing approximative heuristics for GED from existing techniques for QAP. In this paper, we focus on the case where the edit costs that underlie GED are quasimetric. This is the case in many applications of GED. We show that, for quasimetric edit costs, it is possible to reduce the size of the corresponding QAP formulation. An empirical evaluation shows that this reduction significantly speeds up the QAP-based approximative heuristics for GED.
							</p>
							<p>
								<a href="https://doi.org/10.1109/ICPR.2018.8546055" target="_blank">doi</a> | <a href="res/icpr2018-poster.pdf" target="_blank">poster</a> | <a href="res/icpr2018.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- SSPR 2018 -->
				<div class="wrap-collabsible left-align">
					<input id="sspr2018" class="toggle" type="checkbox">
					<label for="sspr2018" class="lbl-toggle">
						<strong>[2018]</strong> D. B. Blumenthal, S. Bougleux, J. Gamper, L. Brun:
						&ldquo;Ring based approximation of graph edit distance&rdquo;,
						<strong>S+SSPR 2018</strong>, pp. 293-303.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> The graph edit distance (GED) is a flexible graph dissimilarity measure widely used within the structural pattern recognition field. A widely used paradigm for approximating GED is to define local structures rooted at the nodes of the input graphs and use these structures to transform the problem of computing GED into a linear sum assignment problem with error correction (LSAPE). In the literature, different local structures such as incident edges, walks of fixed length, and induced subgraphs of fixed radius have been proposed. In this paper, we propose to use rings as local structure, which are defined as collections of nodes and edges at fixed distances from the root node. We empirically show that this allows us to quickly compute a tight approximation of GED.
							</p>
							<p>
								<a href="https://doi.org/10.1007/978-3-319-97785-0_28" target="_blank">doi</a> | <a href="res/sspr2018-slides.pdf" target="_blank">slides</a> | <a href="res/sspr2018.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- TKDE 2018 -->
				<div class="wrap-collabsible left-align">
					<input id="tkde2018" class="toggle" type="checkbox">
					<label for="tkde2018" class="lbl-toggle">
						<strong>[2018]</strong> D. B. Blumenthal, J. Gamper:
						&ldquo;Improved lower bounds for graph edit distance&rdquo;,
						<strong>IEEE Trans. Knowl. Data Eng. 30(3)</strong>, pp. 503-516.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> The problem of deriving lower and upper bounds for the edit distance between undirected, labelled graphs has recently received increasing attention. However, only one algorithm has been proposed that allegedly computes not only an upper but also a lower bound for non-uniform edit costs and incorporates information about both node and edge labels. In this paper, we demonstrate that this algorithm is incorrect. We present a corrected version <span class="smallcaps">Branch</span> that runs in O(n<sup>2</sup>&Delta;<sup>3</sup>+n<sup>3</sup>) time, where &Delta; is the maximum of the maximum degrees of input graphs G and H. We also develop a speed-up <span class="smallcaps">BranchFast</span> that runs in O(n<sup>2</sup>&Delta;<sup>2</sup>+n<sup>3</sup>) time and computes an only slightly less accurate lower bound. The lower bounds produced by <span class="smallcaps">Branch</span> and <span class="smallcaps">BranchFast</span> are shown to be pseudo-metrics on a collection of graphs. Finally, we suggest an anytime algorithm <span class="smallcaps">BranchTight</span> that iteratively improves <span class="smallcaps">Branch</span>'s lower bound. <span class="smallcaps">BranchTight</span> runs in O(n<sup>3</sup>&Delta;<sup>2</sup>+I(n<sup>2</sup>&Delta;<sup>3</sup>+n<sup>3</sup>)) time, where the number of iterations I is controlled by the user. A detailed experimental evaluation shows that all suggested algorithms are Pareto optimal, that they are very effective when used as filters for edit distance range queries, and that they perform excellently when used within classification frameworks.
							</p>
							<p>
								<a href="https://doi.org/10.1109/TKDE.2017.2772243" target="_blank">doi</a> | <a href="res/tkde2018.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- GbRPR 2017 -->
				<div class="wrap-collabsible left-align">
					<input id="gbrpr2017" class="toggle" type="checkbox">
					<label for="gbrpr2017" class="lbl-toggle">
						<strong>[2017]</strong> D. B. Blumenthal, J. Gamper:
						&ldquo;Exact computation of graph edit distance for uniform and non-uniform metric edit costs&rdquo;,
						<strong>GbRPR 2017</strong>, pp. 211-221. <strong>Best Paper Award</strong>.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> The graph edit distance is a well-established and widely used distance measure for labelled, undirected graphs. However, since its exact computation is NP-hard, research has mainly focused on devising approximative heuristics and only few exact algorithms have been proposed. The standard approach <span class="typewriter">A&#42;-GED</span>, a node-based best-first search that works for both uniform and non-uniform metric edit costs, suffers from huge runtime and memory requirements. Recently, two better performing algorithms have been proposed: <span class="typewriter">DF-GED</span>, a node-based depth-first search that works for uniform and non-uniform metric edit costs, and <span class="typewriter">CSI_GED</span>, an edge-based depth-first search that works only for uniform edit costs. Our paper contains two contributions: First, we propose a speed-up <span class="typewriter">DF-GED<sup>u</sup></span> of <span class="typewriter">DF-GED</span> for uniform edit costs. Second, we develop a generalisation <span class="typewriter">CSI_GED<sup>nu</sup></span> of <span class="typewriter">CSI_GED</span> that also covers non-uniform metric edit cost. We empirically evaluate the proposed algorithms. The experiments show, i.a., that our speed-up <span class="typewriter">DF-GED<sup>u</sup></span> clearly outperforms <span class="typewriter">DF-GED</span> and that our generalisation <span class="typewriter">CSI_GED<sup>nu</sup></span> is the most versatile algorithm.
							</p>
							<p>
								<a href="https://doi.org/10.1007/978-3-319-58961-9_19" target="_blank">doi</a> | <a href="res/gbrpr2017-slides.pdf" target="_blank">slides</a> | <a href="res/gbrpr2017.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div>
				<!-- ICDE 2017 -->
				<div class="wrap-collabsible left-align">
					<input id="icde2017" class="toggle" type="checkbox">
					<label for="icde2017" class="lbl-toggle">
						<strong>[2017]</strong> D. B. Blumenthal, J. Gamper:
						&ldquo;Correcting and speeding-up bounds for non-uniform graph edit distance&rdquo;,
						<strong>ICDE 2017</strong>, pp. 131-134.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> The problem of deriving lower and upper bounds for the edit distance between labelled undirected graphs has recently received increasing attention. However, only one algorithm has been proposed that allegedly computes not only an upper but also a lower bound for non-uniform metric edit costs and incorporates information about both node and edge labels. In this paper, we show that this algorithm is incorrect in the sense that, in general, it does not compute a lower bound. We present <span class="smallcaps">Branch</span>, a corrected version of the algorithm that runs in O(n<sup>5</sup>) time. We also develop a speed-up <span class="smallcaps">BranchFast</span> that runs in O(n<sup>4</sup>) time and computes a lower bound, which is only slightly less accurate than the one computed by <span class="smallcaps">Branch</span>. An experimental evaluation shows that <span class="smallcaps">Branch</span> and <span class="smallcaps">BranchFast</span> yield excellent runtime/accuracy-tradeoffs, as they outperform all existing competitors in terms of runtime or in terms of accuracy.
							</p>
							<p>
								<a href="https://doi.org/10.1109/ICDE.2017.57" target="_blank">doi</a> | <a href="res/icde2017-poster.pdf" target="_blank">poster</a> | <a href="res/icde2017.bib" target="_blank">bibtex</a>
							</p>
						</div>
					</div>
				</div> 
				<!-- GAP 2012 -->
				<div class="wrap-collabsible left-align">
					<input id="gap2012" class="toggle" type="checkbox">
					<label for="gap2012" class="lbl-toggle">
						<strong>[2012]</strong> D. B. Blumenthal:
						&ldquo;Rede &uuml;ber fiktive Kontexte&rdquo;,
						<strong>GAP.8 2012</strong>, pp. 16-31.
					</label>
					<div class="collapsible-content">
						<div class="content-inner">
							<p>
								<strong>Abstract</strong> Der Aufsatz zielt darauf ab, ein angemessenes Verst&auml;ndnis der Semantik von &Auml;ußerungen &uuml;ber fiktive Kontexte (kurz: AFKs) zu entwickeln. Der systematische Ausgangspunkt der Arbeit besteht dabei in einem pragmatistischen Inferentialismus &agrave; la Robert Brandom. F&uuml;r diese theoretische Weichenstellung wird nicht gesondert argumentiert, sondern vielmehr geltend gemacht, dass aus ihr zwei Forderungen an eine angemessene Theorie der Semantik von AFKs ableitbar sind. Erstens muss eine solche Theorie den inferentiellen Beziehungen Rechnung tragen, in denen von AFKs gemachte Aussagen de facto stehen. Zweitens darf sie nur auf solche Gegenst&auml;nde zur&uuml;ckgreifen, die insofern ontologisch unschuldig sind, als sie vollst&auml;ndig und individuierbar sind. Aus diesen Forderungen ergibt sich, dass klassische Theorien der Semantik von AFKs unbefriedigend sind: Weder können AFKs mit Bertrand Russell als &Auml;ußerungen &uuml;ber das inferentielle Potenzial bestimmter Mengen fiktionaler Medien betrachtet, noch mit Searle, van Inwagen oder Parsons als &Auml;ußerungen &uuml;ber fiktive Gegenst&auml;nde verstanden werden. Im Anschluss an diese kritische Auseinandersetzung wird ein eigener Vorschlag entwickelt, dessen Kerngedanken darin bestehen, AFKs als &Auml;ußerungen &uuml;ber bestimmte Werke zu betrachten, die wesentlich auf Interpretation beruhen. Dabei werden Werke als &Auml;quivalenzklassen fiktionaler Medien verstanden, die logische Feinstruktur von AFKs gemachter Aussagen nach dem Modell von De-dicto-Zuschreibungen erl&auml;utert und deren Funktionsweise wiederum inferentialistisch gefasst.
							</p>
							<p>
								<a href="http://duepublico.uni-duisburg-essen.de/servlets/DerivateServlet/Derivate-33085/GAP8_Proceedings.pdf/" target="_blank">pdf</a> | <a href="res/gap2012.bib" target="_blank">bibtex</a> 
							</p>
						</div>
					</div>
				</div>
			</div>

			<!--- Teaching --->
			<div class="docs-section" id="teaching">
				<h6 class="docs-header">Teaching</h6>
				<dl class="left-align">
					<dt><strong>Technical University of Munich</strong></dt>
					<dd>
						Einf&uuml;hrung in die Bioinformatik I (2020/2021, labs and exams, B.Sc. in Bioinformatics), Einf&uuml;hrung in die Bioinformatik II, (2020, labs and exams, B.Sc. in Bioinformatics).
					</dd>
					<dt><strong>Free University of Bozen-Bolzano</strong></dt>
					<dd>
						Mathematical Methods for Experimental Science (2017/2018, labs, B.Sc. in Computer Science), Data Structures and Algorithms (2016/2017, labs, B.Sc. in Computer Science), Calculus (2015/2016, labs, B.Sc. in Computer Science).
					</dd>
					<dt><strong>Supervised Master Students</strong></dt>
					<dd>
						M. Hoffmann (2020, M.Sc. in Bioinformatics), K. Adamowicz (2020/2021, M.Sc. in Bioinformatics).
					</dd>
					<dt><strong>Supervised Bachelor Students</strong></dt>
					<dd>
						R. Reitmeir (2020/2021, B.Sc. in Bioinformatics), T. Mauermeier (2020/2021, B.Sc. in Bioinformatics).
					</dd>
				</dl>
			</div>

			<!-- Software -->
			<div class="docs-section" id="software">
				<h6 class="docs-header">Software</h6>
				<div class="row">
					<div class="two columns">
						<h6><a class="project-link" href="https://www.exbio.wzw.tum.de/covex/" target="_blank">CoVex</a></h6>
					</div>
					<div class="ten columns left-align">
						<p>
							<a href="https://www.exbio.wzw.tum.de/covex/" target="_blank">CoVeX</a> (Coronavirus Explorer) is a unique online network and systems medicine platform for data analysis that integrates virus-human interactions for SARS-CoV-2 and SARS-CoV-1. It implements different network-based approaches for the identification of new drug targets and new repurposable drugs. For detailed information please visit our <a href="https://www.exbio.de/exbio-vs-covid-part-1/" target="_blank">blog</a>.
						<p>
					</div>
				</div>
				<div class="row">
					<div class="two columns">
						<h6><a class="project-link" href="https://github.com/dbblumenthal/gedlib" target="_blank">GEDLIB</a></h6>
					</div>
					<div class="ten columns left-align">
						<p>
							<a href="https://github.com/dbblumenthal/gedlib" target="_blank">GEDLIB</a> is an easily extensible C++ library for (suboptimally) computing the graph edit distance (GED) between two labeled graphs. GED is defined as the minimum cost of a sequence of elementary edit operations that transforms one graph into another. There are six elementary edit operations: changing the label of an existing node, inserting an isolated node, deleting an isolated node, changing the label of an existing edge, inserting an an edge between two existing nodes, and deleting an edge. GEDLIB implements several state of the art methods for computing GED and comes with predefined edit costs for some benchmark datasets. Further methods and edit costs can easily be implemented by the user.
						<p>
					</div>
				</div>
				<div class="row">
					<div class="two columns">
						<h6><a class="project-link" href="https://github.com/baumbachlab/epigen" target="_blank">EpiGEN</a></h6>
					</div>
					<div class="ten columns left-align">
						<p>
							<a href="https://github.com/baumbachlab/epigen" target="_blank">EpiGEN</a> is an easy-to-use epistasis simulation pipeline written in Python. It supports epistasis models of arbitrary size, which can be specified either extensionally or via parametrized risk models. Moreover, the user can specify the minor allele frequencies (MAFs) of both noise and disease SNPs, and provide a bias target distribution for the generated phenotypes to simulate observation bias.
						<p>
					</div>
				</div>
			</div>

			<!-- Awards & Funding -->
			<div class="docs-section" id="awards">
				<h6 class="docs-header">Awards & Funding</h6>
				<dl class="left-align">
					<dt><strong>2015 - 2019</strong></dt>
					<dd>
						Ph.D. grant from the <a href="http://www.inf.unibz.it/">Faculty of Computer Science</a> of the <a href="http://www.unibz.it/">Free University of Bozen-Bolzano</a>.
					</dd>
					<dt><strong>2017</strong></dt> 
					<dd>
						Best paper award for <a href = "https://doi.org/10.1007/978-3-319-58961-9_19" target="_blank">&ldquo;Exact computation of graph edit distance for uniform and non-uniform metric edit costs&rdquo;</a> at <a href="http://gbr2017.unisa.it/site/">GbRPR 2017</a>.
					</dd>
					<dt><strong>2015</strong></dt>
					<dd>
						M.Sc. award from the <a href="http://www.math.tu-berlin.de/menue/home/">Institue of Mathematics</a> of the <a href="http://www.tu-berlin.de/">Technische Universit&auml;t Berlin</a>.
					</dd>
					<dt><strong>2007 - 2015</strong></dt> 
					<dd>
						Scholarship from the <a href="https://www.studienstiftung.de">Studienstiftung des deutschen Volkes</a>.
					</dd>
				</dl>
			</div>

			<!-- Services -->
			<div class="docs-section" id="services">
				<h6 class="docs-header">Services</h6>
				<dl class="left-align">
					<dt><strong>PC Member</strong></dt>
					<dd>
						HCC 2019.
					</dd>
					<dt><strong>Reviewing for Journals</strong></dt>
					<dd>
						BioData Mining, European Journal of Operational Research, Information Sciences, Information Systems Frontiers, Nature Computational Science, Pattern Analysis and Applications, Pattern Recognition Letters.
					</dd>
					<dt><strong>Reviewing for Conferences</strong></dt> 
					<dd>
						ICPR 2018, VLDB 2017, EDBT 2017, ICDE 2016.
					</dd>
				</dl>
			</div>
	
			<!-- Footer -->
			<section class="footer">
				&copy;2018 D. B. Blumenthal. Designed with <a href="http://getskeleton.com/">Skeleton</a>.
			</section>

		</div>

		<!-- Google Analytics -->
		<script>
			(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
			(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
			m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
			})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
			ga('create', 'UA-103239654-1', 'auto');
			ga('send', 'pageview');
		</script>

	</body>

</html>